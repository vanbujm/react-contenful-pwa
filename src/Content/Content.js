import React from 'react';
import { connect } from 'react-redux';

import { actionCreators } from '../actions';

import './Content.css';

const { content: fetchContent } = actionCreators;

const splitToLocale = (content) => (locales, key) => {
  const data = content[key];
  const localesForKey = Object.keys(data);
  localesForKey.forEach((locale) => {
    locales[locale] = { ...locales[locale], [key]: content[key][locale] };
  });
  return locales;
};

const parseFields = (key, value, locale) => {
  if (typeof value === 'string') return value;
  switch (key) {
    case 'image':
      const localeWithDefault = value.fields.file[locale] === undefined ? value.fields.file[Object.keys(value.fields.file)[0]] : locale;
      const url = value.fields.file[localeWithDefault].url;
      const title = value.fields.title[localeWithDefault];
      return <img className="content-img" src={url} alt={title}/>;
    case 'duration':
      return value;
    default:
      return '';
  }
};

const fieldsFor = (contentObj, locale) => Object.keys(contentObj).map(
  (key) => {
    if (typeof contentObj[key] !== 'string') console.log(key, contentObj[key]);
    return <div className="content-field"
                key={key}>
      <span style={{ fontWeight: 'bold' }}>{key}: </span>
      {parseFields(key, contentObj[key], locale)}
    </div>;
  });

export const Content = (props) => {
  const { content } = props;
  if (Object.keys(content).length === 0) props.fetchContent();

  const localeSplitData = Object.keys(content).reduce(splitToLocale(content), {});

  const fields = Object.keys(localeSplitData).map(locale => <div key={locale}>
    <h2>{locale}</h2>
    {fieldsFor(localeSplitData[locale], locale)}
  </div>);

  return <div>
    <h1>Content</h1>
    <div>{fields}</div>
  </div>;
};

const mapDispatchToProps = {
  fetchContent
};

const mapState = (state) => ({ content: state.content.fields || {} });

export default connect(mapState, mapDispatchToProps)(Content);