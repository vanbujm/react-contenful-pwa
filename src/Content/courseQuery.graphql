query courseQuery {
    courses(where: { courseCode: 1 }) {
      title
      locale
      image {
        url
        altText
      }
      shortDescription
      description
      duration
      skillLevel
    }
}