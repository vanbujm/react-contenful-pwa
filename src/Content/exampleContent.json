{
  "title": {
    "de-DE": "Hallo SDKs",
    "en-US": "Hello SDKs"
  },
  "slug": {
    "en-US": "hello-sdks"
  },
  "image": {
    "en-US": {
      "sys": {
        "space": {
          "sys": {
            "type": "Link",
            "linkType": "Space",
            "id": "tw4ymfsmbef6"
          }
        },
        "id": "6nvWJT1AkM64so8Auue4QQ",
        "type": "Asset",
        "createdAt": "2018-11-05T01:24:46.652Z",
        "updatedAt": "2018-11-05T01:24:46.652Z",
        "environment": {
          "sys": {
            "id": "master",
            "type": "Link",
            "linkType": "Environment"
          }
        },
        "revision": 1
      },
      "fields": {
        "title": {
          "en-US": "Contentful hero"
        },
        "file": {
          "en-US": {
            "url": "//images.ctfassets.net/tw4ymfsmbef6/6nvWJT1AkM64so8Auue4QQ/328f902b490337ec0dd391e202c173ef/EngineeringHistory.png",
            "details": {
              "size": 100983,
              "image": {
                "width": 1908,
                "height": 887
              }
            },
            "fileName": "EngineeringHistory.png",
            "contentType": "image/png"
          }
        }
      }
    }
  },
  "shortDescription": {
    "de-DE": "Lernen Sie den Umgang mit unseren SDKs.",
    "en-US": "Learn about best practices when using our SDKs."
  },
  "description": {
    "de-DE": "Sobald Sie sich den Quellcode der Besipielanwendung ansehen, werden Sie ein Gespür entwickeln, wie Sie am besten die Contentful SDKs in Ihrer Programmiersprache benutzen. Für weitere, fremde Abhängigkeiten, schauen Sie bitte in der jewiligen Dokumentation nach.",
    "en-US": "By looking at the code of the example app, you'll get a sense of how to use a Contentful SDK in your favorite programming language."
  },
  "duration": {
    "en-US": 5
  },
  "skillLevel": {
    "en-US": "beginner"
  }
}