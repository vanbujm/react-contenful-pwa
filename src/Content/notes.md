### Notes
This is the main component the renders contentful content 
