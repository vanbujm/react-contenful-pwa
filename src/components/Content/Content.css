.content-field {
    margin-bottom: 1rem;
}

.content-img{
    width:100%;
    max-width:600px;
    margin: 0.5rem;
}