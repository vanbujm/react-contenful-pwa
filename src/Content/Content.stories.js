import React from 'react';
import { storiesOf } from '@storybook/react';

import { Content } from './Content';
import exampleContent from './exampleContent';
import contentNotes from './notes.md';

storiesOf('Content', module).add(
  'with some graphcms content',
  () => <Content content={exampleContent} />,
  {
    info: {
      text: contentNotes
    }
  }
);
