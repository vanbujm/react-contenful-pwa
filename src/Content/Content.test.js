import React from 'react';
import { Content } from './Content';
import { shallow } from 'enzyme';

import exampleContenfulContent from './exampleContent';

it('Renders content', () => {
  const tree = shallow(<Content content={exampleContenfulContent} />);
  expect(tree).toMatchSnapshot();
});
