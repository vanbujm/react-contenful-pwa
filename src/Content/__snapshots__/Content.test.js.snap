// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders children 1`] = `
<div>
  <h1>
    Content
  </h1>
  <div>
    <div>
      <h2>
        de-DE
      </h2>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          title
          : 
        </span>
        Hallo SDKs
      </div>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          shortDescription
          : 
        </span>
        Lernen Sie den Umgang mit unseren SDKs.
      </div>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          description
          : 
        </span>
        Sobald Sie sich den Quellcode der Besipielanwendung ansehen, werden Sie ein Gespür entwickeln, wie Sie am besten die Contentful SDKs in Ihrer Programmiersprache benutzen. Für weitere, fremde Abhängigkeiten, schauen Sie bitte in der jewiligen Dokumentation nach.
      </div>
    </div>
    <div>
      <h2>
        en-US
      </h2>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          title
          : 
        </span>
        Hello SDKs
      </div>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          slug
          : 
        </span>
        hello-sdks
      </div>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          image
          : 
        </span>
        <img
          alt="Contentful hero"
          className="content-img"
          src="//images.ctfassets.net/tw4ymfsmbef6/6nvWJT1AkM64so8Auue4QQ/328f902b490337ec0dd391e202c173ef/EngineeringHistory.png"
        />
      </div>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          shortDescription
          : 
        </span>
        Learn about best practices when using our SDKs.
      </div>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          description
          : 
        </span>
        By looking at the code of the example app, you'll get a sense of how to use a Contentful SDK in your favorite programming language.
      </div>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          duration
          : 
        </span>
        5
      </div>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          skillLevel
          : 
        </span>
        beginner
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders content 1`] = `
<div>
  <h1>
    Content
  </h1>
  <div>
    <div>
      <h2>
        de-DE
      </h2>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          title
          : 
        </span>
        Hallo SDKs
      </div>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          shortDescription
          : 
        </span>
        Lernen Sie den Umgang mit unseren SDKs.
      </div>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          description
          : 
        </span>
        Sobald Sie sich den Quellcode der Besipielanwendung ansehen, werden Sie ein Gespür entwickeln, wie Sie am besten die Contentful SDKs in Ihrer Programmiersprache benutzen. Für weitere, fremde Abhängigkeiten, schauen Sie bitte in der jewiligen Dokumentation nach.
      </div>
    </div>
    <div>
      <h2>
        en-US
      </h2>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          title
          : 
        </span>
        Hello SDKs
      </div>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          slug
          : 
        </span>
        hello-sdks
      </div>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          image
          : 
        </span>
        <img
          alt="Contentful hero"
          className="content-img"
          src="//images.ctfassets.net/tw4ymfsmbef6/6nvWJT1AkM64so8Auue4QQ/328f902b490337ec0dd391e202c173ef/EngineeringHistory.png"
        />
      </div>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          shortDescription
          : 
        </span>
        Learn about best practices when using our SDKs.
      </div>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          description
          : 
        </span>
        By looking at the code of the example app, you'll get a sense of how to use a Contentful SDK in your favorite programming language.
      </div>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          duration
          : 
        </span>
        5
      </div>
      <div
        className="content-field"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          skillLevel
          : 
        </span>
        beginner
      </div>
    </div>
  </div>
</div>
`;
